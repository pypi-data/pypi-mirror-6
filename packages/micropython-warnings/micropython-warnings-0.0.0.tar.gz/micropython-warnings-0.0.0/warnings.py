# MicroPython doesn't support warnings
