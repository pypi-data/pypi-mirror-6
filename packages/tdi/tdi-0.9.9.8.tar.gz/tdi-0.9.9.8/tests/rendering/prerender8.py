#!/usr/bin/env python
import warnings as _warnings
_warnings.resetwarnings()
_warnings.filterwarnings('error')

from tdi import html

template = html.from_string("""
<x tdi="xnode">
    <y tdi="ynode">
        <z tdi="znode">
        </z>
    </y>
</x>
<a tdi=":xnode">
    <b tdi="bnode">
    </b>
</a>
""".lstrip())

class Model(object):

    def render_xnode(self, node):
        node.repeat(None, u"abc")

    def render_znode(self, node):
        node.content = node.ctx[1]

    def render_bnode(self, node):
        node.content = u', '.join(node.ctx[1])


template.render(prerender=Model())
