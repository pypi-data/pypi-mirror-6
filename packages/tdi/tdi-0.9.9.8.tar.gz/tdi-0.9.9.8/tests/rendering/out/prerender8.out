<x>
    <y>
        <z>a</z>
    </y>
</x><a>
    <b>a, b</b>
</a><x>
    <y>
        <z>b</z>
    </y>
</x><a>
    <b>b, c</b>
</a><x>
    <y>
        <z>c</z>
    </y>
</x>

